pallets/hyperbridge_aggregations/src/benchmarking.rs

```diff
@@ -0,0 +1,134 @@
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use super::*;
+
+use frame_benchmarking::v2::*;
+use frame_support::traits::fungible::Inspect;
+use frame_system::RawOrigin;
+use hex_literal::hex;
+use ismp::host::StateMachine;
+use sp_runtime::traits::Zero;
+
+type BalanceOf<T> = <<T as pallet_ismp::Config>::Currency as Inspect<
+    <T as frame_system::Config>::AccountId,
+>>::Balance;
+
+pub mod utils {
+    use super::*;
+    use frame_support::traits::fungible::Mutate;
+    use sp_runtime::traits::Bounded;
+
+    pub static DEFAULT_EMPTY_ATT: [u8; 32] =
+        hex!("290decd9548b62a8d60345a988386fc84ba6bc95484008f6362f93160ef3e563");
+
+    pub static TEST_CONTRACT: [u8; 20] = hex!("d9145CCE52D386f254917e481eB44e9943F39138");
+
+    /// Return a whitelisted account with enough founds to do anything.
+    pub fn funded_account<T: Config>() -> T::AccountId {
+        let caller: T::AccountId = whitelisted_caller();
+        T::Currency::set_balance(&caller, BalanceOf::<T>::max_value() / 2u32.into());
+        caller
+    }
+}
+
+#[benchmarks]
+mod benchmarks {
+    use super::utils::*;
+    use super::*;
+    use sp_runtime::traits::Bounded;
+    use sp_runtime::Saturating;
+
+    const fn get_max_benchmark_iterations() -> u32 {
+        1_000_000
+    }
+
+    #[benchmark]
+    fn dispatch_aggregation() {
+        let caller: T::AccountId = funded_account::<T>();
+
+        // Prepare the parameters
+        let params = Params {
+            aggregation_id: 1u64,
+            aggregation: sp_core::H256(DEFAULT_EMPTY_ATT),
+            module: sp_core::H160(TEST_CONTRACT),
+            destination: StateMachine::Kusama(4009),
+            timeout: 0,
+            fee: Zero::zero(),
+        };
+
+        let initial_nonce = pallet_ismp::Nonce::<T>::get();
+        let initial_event_count = frame_system::Pallet::<T>::events().len();
+
+        #[extrinsic_call]
+        dispatch_aggregation(RawOrigin::Signed(caller), params);
+
+        // Assertions
+        assert!(
+            frame_system::Pallet::<T>::events().len() > initial_event_count,
+            "No events were emitted"
+        );
+        assert_eq!(
+            pallet_ismp::Nonce::<T>::get(),
+            initial_nonce + 1,
+            "Nonce should be incremented"
+        );
+    }
+
+    #[benchmark]
+    fn dispatch_aggregation_var_fee(n: Linear<1, { get_max_benchmark_iterations() }>) {
+        let caller: T::AccountId = funded_account::<T>();
+
+        let max_fee = BalanceOf::<T>::max_value() / 4u32.into();
+        let fee = max_fee.saturating_mul(n.into()) / get_max_benchmark_iterations().into();
```

Comment:
I believe this is not actually doing what you intended: when  `n` grows past `4`, `max_fee` saturates to `BalanceOf::<T>::max_value()` before the division.

Besides, it should be the benchmark framework that uses a given number of iterations, so you should not care about that directly in your code. In other words, I believe that in this code, `n`  should only specify the valid domain for `fee` (e.g. `Linear<0, { max_fee() }>` with `max_fee()` returning `BalanceOf::<T>::max_value() / 4u32.into()`)

Reply:
Variability is minimal, but we need to depict the worst case here: let's set some timeout and fee (I don't think it's relevant the timeout and fee amount)

Reply:
Ok